python_modules/dagster/dagster/_core/pipes/context.py

```diff
@@ -191,6 +194,16 @@ def _handle_log(self, message: str, level: str = "info") -> None:
         check.str_param(message, "message")
         self._context.log.log(level, message)
 
+    def report_component_exception(self, origin: str, exc_info: ExceptionInfo):
+        # report as an engine event to provide structured exception data
+        DagsterEvent.engine_event(
+            self._context.get_step_execution_context(),
+            f"[pipes] exception occurred in component {origin}",
+            EngineEventData(
+                error=serializable_error_info_from_exc_info(exc_info),
+            ),
+        )
```

Comment:
i tried to change the other "system" pipes message to engine event to align this, but the front-end handling for system events doesn't handle warning and only errors if you have the exception payload. I did a quick stab at updating the front-end and engine event code to support "warnings" but it still looked scuffed so bailed back to this.